keps/sig-apps/2214-indexed-job/README.md

```diff
@@ -206,16 +207,28 @@ type JobSpec struct {
   // `NonIndexed`.
   CompletionMode CompletionMode
 }	
+
+type JobStatus struct {
+  ...
+
+  // CompletedIndexes holds the completed indexes when .spec.completionMode =
+  // "Indexed" in a text format. The indexes are represented as decimal integers
+  // separated by commas. The numbers are listed in increasing order. Two or
+  // more consecutive numbers are compressed and represented by the first and
+  // last element of the series, separated by a hyphen.
+  // For example, if the completed indexes are 1, 3, 4, 5 and 7, they are
+  // represented as "1,3-5,7".
+  CompletedIndexes string
+}
 ```
 
-As the comment describes, when `.spec.completionMode = "Indexed"`, the
-`.spec.completions` must be:
+As the comment describes, when `.spec.completionMode = "Indexed"`:
 
-- a non-zero positive value. This is to trigger Job management strategy for
-  *fixed completion count*. That is, `Indexed` mode cannot	be used for work
-  queue patterns.	
-- less than or equal to `10^6`. This is to guarantee that we can keep track of
-  completions per-index in the Job status in the future.
+- `.spec.completions` must be a non-zero positive value. This is to trigger Job
+  management strategy for *fixed completion count*. That is, `Indexed` mode
+  cannot be used for work queue patterns.	
+- `.spec.parallelism` must be less than or equal to `10^5`. This is to guarantee
```

Comment:
Actually, I started thinking is 10^5 actually enough?
In the most pathological case, where the indices that were finished are: `1,3,5,7,...,199999'

then if you're saying that you store that as string literally, then you need O(7*10^5) bytes to represent that.
Which actually fits 1.5MB.

So let's take even more pathological case:
`1-999999999, 1000000001, 1000000003, 1000000005, ... `

Then you already need ~1MB to store that. Which still sounds fine.

So actually maybe it's fine - I guess I'm too paranoid here....

Reply:
Answering below